doc/sphinx/proof-engine/ltac.rst

```diff
@@ -707,13 +707,25 @@ First tactic to succeed
 In some cases backtracking may be too expensive.
 
 .. tacn:: first [ {*| @ltac_expr } ]
+          first @ltac_expr
+   :name: first; _
+
+   In the first form: for each focused goal, independently apply the first
+   :token:`ltac_expr` that succeeds.
+
+   In the second form: :n:`@ltac_expr` is an :n:`@ident` that represents a list
```

Comment:
If it is really restricted to an ident, then we could adjust the documented grammar.